evalml/tests/component_tests/detrender_tests/test_polynomial_decomposer.py

```diff
@@ -0,0 +1,400 @@
+        y_shuffled = y.sample(frac=1, replace=False)
+        decomposer.fit_transform(X, y_shuffled)
+    expected_errors = ["monotonically", "X must be in an sktime compatible format"]
+    assert any([error in str(exec_info.value) for error in expected_errors])
+    with pytest.raises(
+        Exception,
+    ):
+        y_string_index = pd.Series(np.arange(31), index=[f"row_{i}" for i in range(31)])
+        decomposer.fit_transform(X, y_string_index)
+
+
+@pytest.mark.parametrize(
+    "frequency",
+    [
+        "D",
+        "W",
+        "S",
+        "h",
+        "T",
+        pytest.param(
+            "m",
+            marks=pytest.mark.xfail(reason="Frequency considered ambiguous by pandas"),
+        ),
+        pytest.param(
+            "M",
+            marks=pytest.mark.xfail(reason="Frequency considered ambiguous by pandas"),
+        ),
+        pytest.param(
+            "Y",
+            marks=pytest.mark.xfail(reason="Frequency considered ambiguous by pandas"),
+        ),
+    ],
+)
+@pytest.mark.parametrize(
+    "test_first_index",
+    ["on period", "before period", "just after period", "mid period"],
+)
+def test_polynomial_decomposer_build_seasonal_signal(
+    ts_data,
+    test_first_index,
+    frequency,
+):
+    period = 7
+    test_first_index = {
+        "on period": 3 * period,
+        "before period": 3 * period - 1,
+        "just after period": 3 * period + 1,
+        "mid period": 3 * period + 4,
+    }[test_first_index]
+
+    # Data spanning 2020-10-01 to 2020-10-31
+    X, y = ts_data
+
+    # Change the date time index to start at the same time but have different frequency
+    y.set_axis(
+        pd.date_range(start="2022-10-01", periods=len(y), freq=frequency),
+        inplace=True,
+    )
+
+    decomposer = PolynomialDecomposer(degree=2)
+
+    # Synthesize a one-week long cyclic signal
+    single_period_seasonal_signal = np.sin(y[0:period] * 2 * np.pi / len(y[0:period]))
+    full_seasonal_signal = np.sin(y * 2 * np.pi / len(y[0:period]))
+
+    # Split the target data.  Since the period of this data is 7 days, we'll test
+    # when the cycle begins, an index before it begins, an index after it begins
+    # and in the middle of a cycle
+    y_test = y[test_first_index:]
+
+    projected_seasonality = decomposer._build_seasonal_signal(
+        y_test,
+        single_period_seasonal_signal,
+        period,
+        frequency,
+    )
+
+    # Make sure that the function extracted the correct portion of the repeating, full seasonal signal
+    assert np.allclose(
+        full_seasonal_signal[projected_seasonality.index],
+        projected_seasonality,
+    )
+
+
+@pytest.mark.parametrize("X_num_time_columns", [0, 1, 2, 3])
+@pytest.mark.parametrize(
+    "X_has_time_index",
+    ["X_has_time_index", "X_doesnt_have_time_index"],
+)
+@pytest.mark.parametrize(
+    "y_has_time_index",
+    ["y_has_time_index", "y_doesnt_have_time_index"],
+)
```

Comment:
This can be cleaned up by using booleans instead of strings (no need to check string equality)

Reply:
I actually do this intentionally so that it shows up in the parametrization list as "y_has_time_index".  It kind of makes it easier to debug if you see FAIL next to all the tests that ran under the "y_has_time_index" parameter vs. "True".

Reply:
<img width="568" alt="image" src="https://user-images.githubusercontent.com/64713315/186953631-36e5637c-c560-4cde-9227-06a1cc04a63c.png">